includes/Generator/Order.php

```diff
@@ -771,23 +771,44 @@ protected static function calculate_refund_totals( $line_items ) {
 
 	/**
 	 * Calculate a realistic refund date based on order completion or previous refund.
+	 * Ensures refund dates never exceed current time and second refunds always occur after first.
 	 *
 	 * @param \WC_Order             $order Order being refunded.
 	 * @param \WC_Order_Refund|null $previous_refund Previous refund (for second refunds).
 	 * @return string Refund date in 'Y-m-d H:i:s' format.
 	 */
 	protected static function calculate_refund_date( $order, $previous_refund = null ) {
+		$now = time();
+
 		if ( $previous_refund ) {
-			// Second refund: within 1 month of first refund
-			$base_date = $previous_refund->get_date_created();
-			$max_days = self::SECOND_REFUND_MAX_DAYS;
+			// Second refund: must be after first refund but before current time
+			$base_timestamp = strtotime( $previous_refund->get_date_created()->date( 'Y-m-d H:i:s' ) );
+			$max_timestamp = min( $base_timestamp + ( self::SECOND_REFUND_MAX_DAYS * DAY_IN_SECONDS ), $now );
+
+			// Ensure second refund is always after first refund
+			if ( $max_timestamp <= $base_timestamp ) {
+				// If there's no time window, use base timestamp + 1 hour (or now if that's in future)
+				$refund_timestamp = min( $base_timestamp + HOUR_IN_SECONDS, $now );
```

Comment:
Does this mean the second refund could occur before the first if the first is in the future?

Reply:
Yeah, I guess if the first refund is in the future, that could happen. Removed the limit in 92130a8fe03eabd37ccb5f0ef4ad12618533106f